src/mlib/int128.test.cpp

```diff
@@ -0,0 +1,305 @@
+#include "./int128.h"
+
+#include <iostream>
+#include <random>
+#include <string>
+
+#if (defined(__GNUC__) && __GNUC__ < 7) || \
+   (defined(_MSC_VER) && _MSC_VER < 1920)
+// Old GCC and old MSVC have partially-broken constexpr that prevents us from
+// properly using static_assert with from_string()
+#define BROKEN_CONSTEXPR
+#endif
```

Comment:
Can the condition used for `_mlibConstexprFn` be used instead for consistency? It is unclear what the overlap / relationship is between the conditions used here vs. those used for `_mlibConstexprFn`.

Reply:
Compilers that match this condition above also declare themselves to have relaxed-constexpr, but it's utterly broken. It would probably be best the other way around, or at least to conditional-ize the "broken" function.

For reference, I whittled it down to this before I realized that the compiler had been gaslighting me:

```c++
// ↓↓↓↓↓ Delete this line, and the static_assert fails: 
constexpr auto completely_unrelated_variable = mlib_int128_mul(NINE, TEN);
// ↑↑↑↑↑ wat ‽

static_assert(mlib_int128_from_string("99999") 
                == mlib_int128(MLIB_INT128_FROM_PARTS(99'999ull, 0)),
              "pain");
```